relay-server/src/services/processor.rs

```diff
@@ -1207,96 +1207,90 @@ impl EnvelopeProcessorService {
         Ok(())
     }
 
-    /// Extract metrics from all envelope items.
-    ///
-    /// Caveats:
-    ///  - This functionality is incomplete. At this point, extraction is implemented only for
-    ///    transaction events.
-    fn extract_metrics(
+    /// Extract transaction metrics.
+    fn extract_transaction_metrics(
```

Comment:
Apologies for the large diff. The rewrite of this function is trying to achieve:

1. Fix the bug referenced in the PR title.
2. Make tandem version bumps unnecessary in the future.
3. Simplify: This function was originally written as a general purpose function, but since it's only called from `process_transactions` we can introduce some early returns.